src/opencl_7z_fmt_plug.c

```diff
@@ -316,17 +316,16 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 	int index;
 	size_t *lws = local_work_size ? &local_work_size : NULL;
 
-	//fprintf(stderr, "%s(%d) lws %zu gws %zu\n", __FUNCTION__, count, local_work_size, global_work_size);
-
 	if (any_cracked) {
 		memset(cracked, 0, cracked_size);
 		any_cracked = 0;
 	}
 
-	if (ocl_autotune_running || new_keys) {
-		int i;
+	global_work_size = GET_NEXT_MULTIPLE(count, local_work_size);
 
-		global_work_size = GET_NEXT_MULTIPLE(count, local_work_size);
+	/* Note: This format is effectively unsalted */
```

Comment:
Why is 7z-opencl "effectively unsalted"?

Reply:
There is a salt field but the salt length is always 0. As long as the iteration count did not change, we don't have to recalculate the decryption keys.

IIRC we recently added some safety checks, so the CPU format *will* handle a future salt correctly while the OpenCL format will reject and warn.